File: 1539-kth-missing-positive-number/1539-kth-missing-positive-number.py
class Solution:
    def findKthPositive(self, arr: List[int], k: int) -> int:
        s,e = 0, len(arr)-1

        while s <= e:
            mid = (s+e)//2
            if arr[mid] - mid - 1 < k: s = mid + 1
            else: e = mid - 1

        return s+k

File: 1539-kth-missing-positive-number/1539-kth-missing-positive-number.ts
function findKthPositive(arr: number[], k: number): number {
    let [s, e] = [0, arr.length-1];

    while(s <= e){
        const mid: number = Math.floor((s + e)/2);
        if((arr[mid] - mid - 1) < k) s = mid + 1;
        else e = mid - 1;
    }

    return s+k;
};